wm/frame.go

```diff
@@ -3,271 +3,214 @@
 package wm
 
 import (
-	"log"
-
 	"fyne.io/fyne"
 	"fyne.io/fyne/app/util"
 	"fyne.io/fyne/test"
 	"fyne.io/fyne/theme"
 
 	"github.com/BurntSushi/xgb/xproto"
-	"github.com/BurntSushi/xgbutil/icccm"
-	"github.com/BurntSushi/xgbutil/xevent"
-	"github.com/BurntSushi/xgbutil/xprop"
 	"github.com/BurntSushi/xgbutil/xwindow"
 
 	"fyne.io/desktop"
 	wmTheme "fyne.io/desktop/theme"
 )
 
 type frame struct {
-	id, win        xproto.Window
-	x, y           int16
-	width, height  uint16
-	mouseX, mouseY int16
-	mouseResize    bool
+	cli                         *client
+	x, y                        int16
+	restoreX, restoreY          int16
+	width, height               uint16
+	restoreWidth, restoreHeight uint16
+	mouseX, mouseY              int16
+	mouseResize                 bool
 
 	framed              bool
 	minWidth, minHeight uint
-	title               string
 
-	wm     *x11WM
 	canvas test.WindowlessCanvas
 }
 
-func (s *stack) frameForWin(id xproto.Window) desktop.Window {
-	for _, w := range s.frames {
-		if w.(*frame).id == id || w.(*frame).win == id {
-			return w
-		}
-	}
-
-	return nil
-}
-
-func (f *frame) unFrame() {
-	if !f.framed {
-		return
-	}
-
-	frame := f.wm.frameForWin(f.win)
-	f.wm.RemoveWindow(frame)
-
-	if frame == nil {
+func (fr *frame) unFrame() {
+	if !fr.framed {
 		return
 	}
 
-	xproto.ReparentWindow(f.wm.x.Conn(), f.win, f.wm.x.RootWin(), f.x, f.y)
-	xproto.UnmapWindow(f.wm.x.Conn(), f.id)
-}
-
-func (f *frame) Decorated() bool {
-	return f.framed
-}
-
-func (f *frame) Title() string {
-	if f.title == "" {
-		f.title = windowName(f.wm.x, f.win)
-	}
-
-	return f.title
-}
-
-func (f *frame) Class() []string {
-	return windowClass(f.wm.x, f.win)
-}
-
-func (f *frame) Command() string {
-	return windowCommand(f.wm.x, f.win)
-}
-
-func (f *frame) IconName() string {
-	return windowIconName(f.wm.x, f.win)
-}
-
-func (f *frame) Close() {
-	winProtos, err := icccm.WmProtocolsGet(f.wm.x, f.win)
-	if err != nil {
-		log.Println("GetProtocols err", err)
-	}
-
-	askNicely := false
-	for _, proto := range winProtos {
-		if proto == "WM_DELETE_WINDOW" {
-			askNicely = true
-		}
-	}
-
-	if !askNicely {
-		err := xproto.DestroyWindowChecked(f.wm.x.Conn(), f.win).Check()
-		if err != nil {
-			log.Println("Close Err", err)
-		}
-
-		return
-	}
+	cli := fr.cli.wm.clientForWin(fr.cli.win)
```

Comment:
This seems strange - isn't that just getting cli from cli's window id???

Reply:
Then why is it called `clientForWin`?

Reply:
what I mean is, looking at the code is `cli` not the same as `fr.cli`?
If they are different this is a naming issue if it's the same then a logic issue.